### ISSUE TITLE
CLI throws exception on VM creation timeout

### ISSUE THREAD

**Issue by sedouard** [Azure/azure-quickstart-templates#1315]
When the deployment fails because of a VM timeout:
```json
{"status":"Failed","error":{"code":"ResourceDeploymentFailure","message":"The resource operation completed with terminal provisioning state 'Failed'.","details":[{"code":"VMStartTimedOut","message":"VM 'hazelcast0' did not start in the allotted time. The VM may still start successfully. Please check the power state later."}]}}
```

The CLI doesn't seem to to handle the error correctly and results in this error:

```
{ [TypeError: Cannot read property 'forEach' of undefined]
  stack: [Getter/Setter],
  __frame: 
   { name: 'exports_getFailedDeploymentOperations__3',
     line: 360,
     file: '/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils._js',
     prev: 
      { name: '__1',
        line: 68,
        file: '/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js',
        prev: [Object],
        calls: 1,
        active: false,
        offset: 10,
        col: 46 },
     calls: 1,
     active: true,
     offset: 4,
     col: 25 },
  rawStack: [Getter] }
TypeError: Cannot read property 'forEach' of undefined
  <<< async stack >>>
    at /app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:100:53
    at Array.forEach (native)
    at /app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:98:84
    at Array.forEach (native)
    at ___ (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:94:46)
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
    at ___ (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils._js:490:16)
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
    at getFailedNestedOperations (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils._js:504:14)
  at exports_getFailedDeploymentOperations__3 (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils._js:364:26)
  at __1 (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:78:47)
  at __1 (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:67:25)
  <<< raw stack >>>
    at /app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:100:53
    at Array.forEach (native)
    at /app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:98:84
    at Array.forEach (native)
    at ___ (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/group.deployment._js:94:46)
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
    at ___ (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils._js:490:16)
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
    at __$getFailedNestedOperations (/app/node_modules/azure-scripty/node_modules/azure-cli/lib/commands/arm/group/groupUtils._js:504:14)
    at ___ (/app/node_modules/streamline/lib/callbacks/runtime.js:111:13)
```

I don't really know what build of the CLI this is, because I was provided a random tarball at `https://github.com/Azure/azure-xplat-cli/archive/dev.tar.gz`. I'm assuming these are live master builds and if that's the case it's from just after the 0.9.15 release.

**Comment by singhkay** [comment]
Looks like this is fixed in https://github.com/Azure/azure-xplat-cli/pull/2498

**Comment by sedouard** [comment]
agreed looks that that'll fix it. closed via #2498
